roles/kubevirt_web_ui/tasks/install.yml

```diff
@@ -0,0 +1,131 @@
+---
+- name: Check if kubevirt-web-ui project exists
+  shell: "{{ openshift_client_binary }} projects -q | grep -w kubevirt-web-ui | awk '{ print $1 }'"
+  register: ns
+
+- name: Create kubevirt-web-ui project
+  shell: "{{ openshift_client_binary }} new-project kubevirt-web-ui"
+  when: ns.stdout != "kubevirt-web-ui"
+
+- name: Make temp directory for console templates
+  command: mktemp -d /tmp/console-ansible-XXXXXX
+  register: mktemp
+  changed_when: False
+
+- name: Copy admin client config
+  command: >
+    cp {{ openshift.common.config_base }}/master/admin.kubeconfig {{ mktemp.stdout }}/admin.kubeconfig
+  changed_when: false
+
+- name: Copy console templates to temp directory
+  copy:
+    src: "{{ item }}"
+    dest: "{{ mktemp.stdout }}/{{ item }}"
+  with_items:
+  - "{{ __console_template_file }}"
+  - "{{ __console_config_file }}"
+
+# kubevirt-ansible: oc_configmap from lib_openshift (openshift-ansible) is not present, so rewritten
+# Check if an existing console-config config map exists. If so, use those
+# contents so we don't overwrite changes.
+
+- name: Read the existing console config map
+  shell: "{{ openshift_client_binary }} get configmap console-config -n kubevirt-web-ui -o jsonpath='{.data}' | true"
+  register: console_config_map
+
+- set_fact:
+    existing_config_map_data: "{{ console_config_map.stdout | default({}) }}"
+
+- name: Copy the existing web console config to temp directory
+  copy:
+    content: "{{ existing_config_map_data['console-config.yaml'] }}"
+    dest: "{{ mktemp.stdout }}/{{ __console_config_file }}"
+  when: existing_config_map_data['console-config.yaml'] is defined
+
+- set_fact:
+    # Must have a trailing slash
+    console_picker_developer_console_public_url: "{{ openshift.master.public_console_url }}/"
+  when: (openshift_web_console_enable_context_selector | default(true) | bool) and (openshift_web_console_install | default(true) | bool)
+
+- set_fact: console_cert={{ lookup('file', kubevirt_web_ui_cert) }}
+  when: kubevirt_web_ui_cert is exists
+  changed_when: false
+
+- set_fact: console_key={{ lookup('file', kubevirt_web_ui_key) }}
+  when: kubevirt_web_ui_key is exists
+  changed_when: false
+
+- set_fact: console_ca_cert={{ lookup('file', kubevirt_web_ui_ca) }}
+  when: kubevirt_web_ui_ca is exists
+  changed_when: false
+
+# kubevirt-ansible: yedit from lib_utils (openshift-ansible) is not present, so rewritten
+# Generate a new config when a config map is not defined.
+
+- name: Set web console config properties from inventory variables
+  block:
+    - name: Change consoleBaseAddress
+      lineinfile:
+        path: "{{ mktemp.stdout }}/{{ __console_config_file }}"
+        regexp: "^  consoleBaseAddress:.*$"
```

Comment:
Could use a jinja2 template?

Reply:
Sounds reasonable considering the code in isolation.
But this change would complicate things when merging with potentially existing ConfigMap. More importantly, I would farther deviate from the openshift-ansible implementation.

Please note, the `kubevirt-web-ui` is a fork of `openshift/console`. For that reason, I would like to stay aligned with their playbook (lives in `openshift-ansible`) as much as possible - it will simplify maintenance in the future.

We work hard to keep both upstream and the fork in sync, as they might be merged in the future. The kubevirt-ansible playbook introduces nothing new compared to the openshift-ansible, just adapts to different project environment.

Reply:
Let's move it to it's own file (`render-config.yaml`) so it's easier to track.  Then you can call the `render-config.yaml` tasks from `install.yaml`.  It will make this more readable so it's easier to maintain.

Reply:
@rthallisey , so the task has been moved to `console-config.yml`.